src/lib/crypto/kimchi_backend/common/plonk_constraint_system.ml

```diff
@@ -1898,93 +1928,113 @@ end = struct
+          |  10 | `carry1_36`   (plookup) | `carry1_72`      (plookup) |
+          |  11 | `carry1_84`             | `carry0`                   |
+          |  12 | `carry1_86`             |                            |
+          |  13 | `carry1_88`             |                            |
+          |  14 | `carry1_90`             |                            |
         *)
+        (* Current row *)
         let vars_curr =
-          [| (* Current row *) Some (reduce_to_v left_input0)
+          [| Some (reduce_to_v left_input0)
            ; Some (reduce_to_v left_input1)
            ; Some (reduce_to_v left_input2)
            ; Some (reduce_to_v right_input0)
            ; Some (reduce_to_v right_input1)
            ; Some (reduce_to_v right_input2)
-           ; Some (reduce_to_v carry1_lo)
-           ; Some (reduce_to_v carry1_hi)
-           ; Some (reduce_to_v carry0)
-           ; Some (reduce_to_v quotient0)
-           ; Some (reduce_to_v quotient1)
-           ; Some (reduce_to_v quotient2)
-           ; Some (reduce_to_v quotient_bound_carry)
-           ; Some (reduce_to_v product1_hi_1)
-           ; None
+           ; Some (reduce_to_v product1_lo)
+           ; Some (reduce_to_v carry1_0)
+           ; Some (reduce_to_v carry1_12)
+           ; Some (reduce_to_v carry1_24)
+           ; Some (reduce_to_v carry1_36)
+           ; Some (reduce_to_v carry1_84)
+           ; Some (reduce_to_v carry1_86)
+           ; Some (reduce_to_v carry1_88)
+           ; Some (reduce_to_v carry1_90)
           |]
         in
+        (* Next row *)
         let vars_next =
-          [| (* Next row *) Some (reduce_to_v remainder0)
-           ; Some (reduce_to_v remainder1)
+          [| Some (reduce_to_v remainder01)
            ; Some (reduce_to_v remainder2)
-           ; Some (reduce_to_v quotient_bound01)
-           ; Some (reduce_to_v quotient_bound2)
-           ; Some (reduce_to_v product1_lo)
+           ; Some (reduce_to_v quotient0)
+           ; Some (reduce_to_v quotient1)
+           ; Some (reduce_to_v quotient2)
+           ; Some (reduce_to_v quotient_hi_bound)
            ; Some (reduce_to_v product1_hi_0)
-           ; None
-           ; None
-           ; None
-           ; None
-           ; None
+           ; Some (reduce_to_v product1_hi_1)
+           ; Some (reduce_to_v carry1_48)
+           ; Some (reduce_to_v carry1_60)
+           ; Some (reduce_to_v carry1_72)
+           ; Some (reduce_to_v carry0)
            ; None
            ; None
            ; None
           |]
         in
-        add_row sys vars_curr ForeignFieldMul [||] ;
+        add_row sys vars_curr ForeignFieldMul
+          [| foreign_field_modulus2
+           ; neg_foreign_field_modulus0
+           ; neg_foreign_field_modulus1
+           ; neg_foreign_field_modulus2
+          |] ;
         add_row sys vars_next Zero [||]
     | Plonk_constraint.T
         (Rot64
-          { (* Current row *) word
+          { word
```

Comment:
Actually it doesnt matter, whether to have them here or not, since berkeley will merge to develop in any case. But having them already in berkeley makes no harm imho.